api/v3rpc/rpctypes/error.go

```diff
@@ -68,19 +68,19 @@ var (
 	ErrGRPCNoLeader                   = status.New(codes.Unavailable, "etcdserver: no leader").Err()
 	ErrGRPCNotLeader                  = status.New(codes.FailedPrecondition, "etcdserver: not leader").Err()
 	ErrGRPCLeaderChanged              = status.New(codes.Unavailable, "etcdserver: leader changed").Err()
-	ErrGRPCNotCapable                 = status.New(codes.Unavailable, "etcdserver: not capable").Err()
+	ErrGRPCNotCapable                 = status.New(codes.FailedPrecondition, "etcdserver: not capable").Err()
 	ErrGRPCStopped                    = status.New(codes.Unavailable, "etcdserver: server stopped").Err()
 	ErrGRPCTimeout                    = status.New(codes.Unavailable, "etcdserver: request timed out").Err()
 	ErrGRPCTimeoutDueToLeaderFail     = status.New(codes.Unavailable, "etcdserver: request timed out, possibly due to previous leader failure").Err()
 	ErrGRPCTimeoutDueToConnectionLost = status.New(codes.Unavailable, "etcdserver: request timed out, possibly due to connection lost").Err()
 	ErrGRPCUnhealthy                  = status.New(codes.Unavailable, "etcdserver: unhealthy cluster").Err()
 	ErrGRPCCorrupt                    = status.New(codes.DataLoss, "etcdserver: corrupt cluster").Err()
-	ErrGPRCNotSupportedForLearner     = status.New(codes.Unavailable, "etcdserver: rpc not supported for learner").Err()
+	ErrGPRCNotSupportedForLearner     = status.New(codes.InvalidArgument, "etcdserver: rpc not supported for learner").Err()
```

Comment:
It seems `Unimplemented` or `Not Found` status is better fit here?

Reply:
How about FailedPrecondition ? 
 - not found - sounds like a problem with 'resource' not 'semantic'. 
 - not implemented - sounds like a way to fix the problem is to upgrade the server. 
 
FailedPrecondition is the most neutral in my opinion

Reply:
`Not implemented` can mean two things:

1. The operation is not implemented
2. The operation is not supported/enabled

This case can fall into the second one.

ref: https://grpc.github.io/grpc/core/md_doc_statuscodes.html.

The precondition failure indicates that the service is in a state that cannot execute the operation. Usually, it is an external triggered state and can be fixed externally (for example to remove an unempty director, client needs to remove all files under it). `learner` is an internal state of the service and cannot be changed by users, but it also makes sense in a border sense.

I have no strong opinion.